subsys/console/tty.c

```diff
@@ -80,6 +89,7 @@ int tty_putchar(struct tty_serial *tty, u8_t c)
 	}
 	if (tx_next == tty->tx_get) {
```

Comment:
is this buffer full case? shouldn't 0 be returned then? btw, any plans to transform it to use ring_buffer?

Reply:
> btw, any plans to transform it to use ring_buffer?

You know that yes (I mentioned that on a couple of occasions). But please my latest comment (in the main comment thread) - the main thing now is to get API right, internals can be refactored then as needed.